apisix/plugins/server-info.lua

```diff
@@ -132,51 +124,111 @@ local function get_server_info()
 end
 
 
+local function set(key, value, ttl)
+    local res_new, err = core.etcd.set(key, value, ttl)
+    if not res_new then
+        core.log.error("failed to set server_info: ", err)
+        return 503, {error_msg = err}
```

Comment:
why should we use `503` here?

Reply:
503 Service Unavailable, I think it is OK to return this error when the plugin cannot provide the service, what do you suggestion?

Reply:
I think `500` is better. `503` is commonly used for overload.